python/pypto/ir/printer.py

```
@@ -26,10 +26,74 @@ def python_print(node: _ir_core.IRNode | _ir_core.Type, prefix: str = "pl") -> s
+    import pypto.language as pl  # noqa: PLC0415
+
+    text = python_print(program)
+
+    try:
+        reparsed = pl.parse(text)
+    except Exception as e:
+        raise ValueError(
+            f"IR roundtrip failed for '{program.name}': re-parsing printed IR raised an error.\n"
+            f"Parse error: {e}\n"
+            f"Printed IR:\n{text}"
+        ) from e
```

Ruff (0.15.4)

[warning] Avoid specifying long messages outside the exception class

(TRY003)